activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -87,14 +88,18 @@ def initialize(secret, *signature_key_or_options)
 
     # Encrypt and sign a message. We need to sign the message in order to avoid
     # padding attacks. Reference: http://www.limited-entropy.com/padding-oracle-attacks.
-    def encrypt_and_sign(value)
+    def encrypt_and_sign(value, expires_at: nil, expires_in: nil, purpose: nil)
+      if expires_at || expires_in || purpose
+        value = Messages::Metadata.wrap(value, expires_at: expires_at, expires_in: expires_in, purpose: purpose)
+      end
       verifier.generate(_encrypt(value))
     end
 
     # Decrypt and verify a message. We need to verify the message in order to
     # avoid padding attacks. Reference: http://www.limited-entropy.com/padding-oracle-attacks.
-    def decrypt_and_verify(value)
-      _decrypt(verifier.verify(value))
+    def decrypt_and_verify(value, purpose: nil)
```

Comment:
Lets rename `value` to `data` — clarifies that this decrypts raw encrypted data.